<a href="https://colab.research.google.com/github/SyedDanishAhmed84/flyrank/blob/main/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:
I choose Lane 2: Refresh / Content Opportunity Scoring. I chose this lane because it addresses a practical content decision: which pages should be reviewed first when time and resources are limited. The starter dataset contains observable signals such as impressions, sessions, content age, freshness, position, CTR, and trend information that can be used to investigate whether pages show measurable signs of refresh opportunity. My goal is not simply to train a model, but to determine whether these signals can support a useful and explainable review priority for content teams.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
### Research question

Can observable content and search-performance signals be used to rank pages by their potential need for review, so that a content team can decide which pages to investigate first?

### Unit of analysis

The unit of analysis will be one content page.

### Output

The intended output is a ranked opportunity score or review queue that helps prioritize pages for human review.

### Decision

The decision is which pages should receive attention first for possible refresh, expansion, protection, pruning, or monitoring.

### Who acts on it

A content or SEO team would use the ranking as decision support when allocating limited review and optimization time.

### Action

A reviewer could inspect the highest-priority pages first and decide whether to refresh, expand, protect, prune, or continue monitoring them.

### Cost of a wrong recommendation

A wrong recommendation could waste limited content resources by prioritizing a page that does not need attention, while a genuinely valuable opportunity could be reviewed later. Therefore, the output should support human judgment rather than automatically determine what action must be taken.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd

df = pd.read_csv("content_refresh_anonymized.csv")

print("Number of rows:", len(df))
print("Number of columns:", len(df.columns))
print("Average impressions:", round(df["impressions_90d"].mean(), 2))

Number of rows: 30000
Number of columns: 44
Average impressions: 5200.37


The starter dataset contains 30,000 observations and 44 columns, providing a substantial set of content and search-performance information to investigate. The average number of 90-day impressions is 5,200.37, showing that the pages in the starter dataset have measurable search exposure. These initial numbers make the Refresh / Content Opportunity Scoring lane worth investigating further over the next seven weeks, while the later analysis will determine whether the available signals are useful for prioritizing pages for human review.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [ ]:
This project can identify and measure patterns in the observable content and search-performance data. The eventual analysis may provide directional insights and decision support for prioritizing pages for human review.

I can discuss observed associations, measured differences, rankings, and patterns in the available data.

I cannot claim that a particular content change will cause a page to recover, improve its rankings, or increase traffic. I also cannot claim to predict Google's ranking system. The available observational data does not provide causal proof. Any score or recommendation should therefore be treated as decision support and validated before being used for important decisions.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.